src/ralph/backends/storage/ldp.py

```diff
@@ -0,0 +1,90 @@
+"""OVH's LDP storage backend for Ralph"""
+
+import logging
+import re
+import sys
+
+import ovh
+import requests
+
+from ..mixins import HistoryMixin
+from .base import BaseStorage
+
+logger = logging.getLogger(__name__)
+
+
+class LDPStorage(HistoryMixin, BaseStorage):
+    """OVH's LDP storage backend"""
+
+    # pylint: disable=too-many-arguments
+
+    name = "ldp"
+
+    def __init__(
+        self,
+        endpoint=None,
+        application_key=None,
+        application_secret=None,
+        consumer_key=None,
+        service_name=None,
+        stream_id=None,
+    ):
+        self.endpoint = endpoint
+        self.application_key = application_key
+        self.application_secret = application_secret
+        self.consumer_key = consumer_key
+        self.service_name = service_name
+        self.stream_id = stream_id
+
+        self.client = ovh.Client(
+            endpoint=self.endpoint,
+            application_key=self.application_key,
+            application_secret=self.application_secret,
+            consumer_key=self.consumer_key,
+        )
+
+    @property
+    def _archive_endpoint(self):
```

Comment:
I don't understand why you are creating a property function here ? This property can be created in the constructor like you did for the `client` property.

Reply:
Because it should be dynamical as it depends on the `service_name` and `stream_id` attributes that may change after `LDPStorage` class instantiation.

Reply:
I see your point @jmaupetit, but I agree with @lunika on this.

How can `service_name` and `stream_id` attributes change after instantiation, but not `application_key`, `application_secret` and `consumer_key` ? They are declared the same way and there is no special setter methods for `service_name` and `stream_id`.

Reply:
As discussed, I have moved client instantiation arguments as private to clarify object usage.